pkg/streamingpromql/types/fpoint_ring_buffer.go

```diff
@@ -58,6 +61,11 @@ func (b *FPointRingBuffer) Append(p promql.FPoint) error {
 			return err
 		}
 
+		if !isPowerOfTwo(cap(newSlice)) {
```

Comment:
Why not enforce this contract in the bucketing pools themselves. ie, don't allow a `factor` and set it to `2`. I think this will also allow us to make looking up the correct bucket for each Get+Put more efficient where we won't need to loop over the buckets.

Reply:
>this will also allow us to make looking up the correct bucket for each Get+Put more efficient where we won't need to loop over the buckets

Yep, I had the same idea 🙂 I'd like to leave that for a separate PR though, this PR is big enough already.

Reply:
Comment added in a91c8b1.